File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Collaborations/CollaborationVersioning.cs
using System;
using VecompSoftware.DocSuiteWeb.Repository.Entity;

namespace VecompSoftware.DocSuiteWeb.Entity.Collaborations
{
    public class CollaborationVersioning : DSWBaseEntity, IEntityLogicDelete
    {
        #region [ Constructor ]

        public CollaborationVersioning() : this(Guid.NewGuid()) { }

        public CollaborationVersioning(Guid uniqueId)
            : base(uniqueId) { }

        #endregion

        #region [ Properties ]

        /// <summary>
        /// Get or set CollaborationIncremental
        /// </summary>
        public short CollaborationIncremental { get; set; }

        /// <summary>
        /// Get or set Incremental
        /// </summary>
        public short Incremental { get; set; }

        /// <summary>
        /// Get or set IdDocument
        /// </summary>
        public int IdDocument { get; set; }

        /// <summary>
        /// Get or set IsActive
        /// </summary>
        public bool IsActive { get; set; }

        /// <summary>
        /// Get or set DocumentName
        /// </summary>
        public string DocumentName { get; set; }

        /// <summary>
        /// Get or set CheckedOut
        /// </summary>
        public bool? CheckedOut { get; set; }

        /// <summary>
        /// Get or set CheckOutUser
        /// </summary>
        public string CheckOutUser { get; set; }

        /// <summary>
        /// Get or set CheckOutSessionId
        /// </summary>
        public string CheckOutSessionId { get; set; }

        /// <summary>
        /// Get or set CheckOutDate
        /// </summary>
        public DateTimeOffset? CheckOutDate { get; set; }

        /// <summary>
        /// Get or set DocumentChecksum
        /// </summary>
        public string DocumentChecksum { get; set; }

        /// <summary>
        /// Get or set DocumentGroup
        /// </summary>
        public string DocumentGroup { get; set; }


        #endregion

        #region [ Navigation Properties ]

        public virtual Collaboration Collaboration { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Collaboration/CollaborationVersioning.vb
<Serializable()>
Public Class CollaborationVersioning
    Inherits AuditableDomainObject(Of Guid)
    Implements ISupportLogicDelete

#Region " Fields "
#End Region

#Region " Properties "

    Public Overridable Property CollaborationIncremental As Short

    Public Overridable Property Incremental As Short

    Public Overridable Property IdDocument As Integer

    Public Overridable Property DocumentName As String

    ''' <summary> </summary>
    ''' <remarks> SUL DB È UN BIT </remarks>
    Public Overridable Property IsActive As Short Implements ISupportLogicDelete.IsActive

    Public Overridable Property Collaboration As Collaboration

    Public Overridable Property IdRepository As String

    Public Overridable Property CheckedOut As Boolean?

    Public Overridable Property CheckOutUser As String

    Public Overridable Property CheckOutSessionId As String

    Public Overridable Property CheckOutDate As DateTimeOffset?

    Public Overridable Property DocumentChecksum As String

    Public Overridable Property DocumentGroup As String


#End Region

#Region " Constructors "

    Public Sub New()
    End Sub

#End Region

End Class


